### PR TITLE
Fixup workflow redux

### PR DESCRIPTION
**Pull request by rikeshi**
Continuation of #320

### CODE REVIEW

File: bioblend/galaxy/objects/wrappers.py
Comment (nsoranzo) [inline bioblend/galaxy/objects/wrappers.py:1456]:
```suggestion
    BASE_ATTRS = Preview.BASE_ATTRS + ('latest_workflow_uuid', 'number_of_steps', 'owner', 'published', 'show_in_tool_panel', 'tags')
```

Comment (nsoranzo) [reply bioblend/galaxy/objects/wrappers.py:1456]: Can you add `latest_workflow_uuid` also to the `Workflow` class above?

File: bioblend/galaxy/invocations/__init__.py
Comment (rikeshi) [reply bioblend/galaxy/invocations/__init__.py:30]: Added it, not sure if the description is precise enough.

File: bioblend/galaxy/objects/wrappers.py
Comment (nsoranzo) [inline bioblend/galaxy/objects/wrappers.py:187]:
```suggestion
        'deleted',
        'inputs',
        'latest_workflow_uuid',
        'owner',
        'published',
        'steps',
        'tags',
```

File: bioblend/galaxy/objects/wrappers.py
Comment (nsoranzo) [inline bioblend/galaxy/objects/wrappers.py:1456]:
```suggestion
    BASE_ATTRS = Preview.BASE_ATTRS + (
        'latest_workflow_uuid',
        'number_of_steps',
        'owner',
        'published',
        'show_in_tool_panel',
        'tags',
    )
```

File: bioblend/galaxy/invocations/__init__.py
Comment (nsoranzo) [inline bioblend/galaxy/invocations/__init__.py:35]:
```suggestion
        :type include_terminal: bool
        :param include_terminal: Whether to include terminal states.

        :type limit: int
        :param limit: Maximum number of invocations in the query result.
```

File: bioblend/galaxy/invocations/__init__.py
Comment (simonbray) [reply bioblend/galaxy/invocations/__init__.py]: It seems so.

Comment (rikeshi) [reply bioblend/galaxy/invocations/__init__.py]: It seems to be the first `limit` invocations ordered by their `id`.

Query: https://github.com/galaxyproject/galaxy/blob/85fafd5a0628b8e127c111c84ce2263678dc220a/lib/galaxy/managers/workflows.py#L253

Limit applied: https://github.com/galaxyproject/galaxy/blob/85fafd5a0628b8e127c111c84ce2263678dc220a/lib/galaxy/managers/workflows.py#L281

Comment (simonbray) [reply bioblend/galaxy/invocations/__init__.py]: If you try `/api/invocations` it's clearly ordered by time though and `/api/invocations?limit=5` shows the most recent ones. Not sure what to make of the code you linked.

Comment (rikeshi) [reply bioblend/galaxy/invocations/__init__.py]: I looked at the index_invocations() of galaxy and what it does with the limit parameter. If the output is ordered by time then I misinterpreted what it does

Comment (simonbray) [reply bioblend/galaxy/invocations/__init__.py]: Yeah, I don't really understand what the code is doing there - as you say it looks like it sorts by ID

Comment (nsoranzo) [reply bioblend/galaxy/invocations/__init__.py]: Ordering by id or create time should be equivalent, since it's order on the autoincrement IDs stored in the database, not the encoded IDs returned by the API, or am I missing something?

Comment (simonbray) [reply bioblend/galaxy/invocations/__init__.py]: No, that makes perfect sense.

File: bioblend/galaxy/invocations/__init__.py
Comment (nsoranzo) [inline bioblend/galaxy/invocations/__init__.py:35]:
```suggestion
        :param limit: Maximum number of invocations to return.
```

Comment (simonbray) [reply bioblend/galaxy/invocations/__init__.py:35]:
```suggestion
        :param limit: Maximum number of invocations to return - if specified, the most recent invocations will be returned.
```

File: bioblend/galaxy/invocations/__init__.py
Comment (nsoranzo) [inline bioblend/galaxy/invocations/__init__.py:19]:
```suggestion
        Get all workflow invocations or those selected by filtering by some
        arguments (e.g. by workflow).
```

Not sure I like this formulation 100%, but other similar methods use something like:

```
        Get all workflow invocations or filter the specific one(s) by workflow
        or other arguments.
```

which I like even less.

Comment (simonbray) [reply bioblend/galaxy/invocations/__init__.py:19]: If you want a third option:
```
Get all workflow invocations, or select a subset by specifying optional arguments for filtering (e.g. a workflow ID)
```

Comment (rikeshi) [reply bioblend/galaxy/invocations/__init__.py:19]: Third option is clearest I think.

Comment (nsoranzo) [reply bioblend/galaxy/invocations/__init__.py]: Thanks @simonbray , I like it! I've adopted it in lieu of the previous formulation for all `get_` methods in commit e16831f3a72b0f2c560e384693faea30fe4566a1 .

**Comment by simonbray** [comment]
Looks okay to me, the only thing that would be nice would be to use the new `wait_invocation` method rather than `time.sleep(20)` in the test. So perhaps we wait until the other PR is merged for that.

**Comment by rikeshi** [comment]
Thanks for fixing the user filtering @simonbray !

**Comment by simonbray** [comment]
You're welcome, it seems changing `gi._key` was not enough to run invocations with the user accounts. But your version worked fine for me locally, I'm not sure what was different on the CI.

**Review by simonbray — APPROVED** [review]

**Review by nsoranzo — APPROVED** [review]

**Comment by nsoranzo** [comment]
Thanks @rikeshi, another great one!

**Pull request merged**
